### PR TITLE
78 related skeleton

### PR DESCRIPTION
**Pull request by jimmyko360**
Widget container for related products only

- Outfits container does not work yet and is commented out in RelatedProducts.jsx
- Renders info for each related product
- clicking on a product updates the state in App.jsx and also updates list of related products
- product images are not updating correctly, asynchronous requests to the API lag significantly behind all other data

### CODE REVIEW

File: client/components/relatedWidget/ArrowButton.jsx [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import React from 'react';
+
+const ArrowButton = ({ text, className }) => {
```
Comment (cchauche) [inline client/components/relatedWidget/ArrowButton.jsx:3]: Nice idea with passing in className to make this a reusable component!

File: client/components/relatedWidget/ProductList.jsx [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import ProductListItem from './ProductListItem.jsx';
+
+const ProductList = (list, selected, callback) => {
```
Comment (cchauche) [inline client/components/relatedWidget/ProductList.jsx:4]: If you are doing object destructuring here shouldn't all the arguments be wrapped in `{}`
so the line would be: 
`const ProductList = ({list, selected, callback}) => {`

File: client/components/relatedWidget/ProductListItem.jsx [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    // console.log('image:', this.props.image);
+    return (<div
+      onClick={()=>{ this.props.getNewProduct(this.props.text.id); }}
+      className={`menu-item ${this.props.selected ? 'active' : ''}`}
+      >
+        <h4>{this.props.text.category}</h4>
+        <h4>{this.props.text.name}</h4>
+        <img src={this.props.image} width='200' height='250'/>
+        <h4>{this.props.text.default_price}</h4>
+        <StarAverage/>
```
Comment (cchauche) [inline client/components/relatedWidget/ProductListItem.jsx:19]: You will eventually have to pass down reviewAverage from App state to this component.
line would be something like: `<StarAverage reviewAverage={this.props.reviewAverage}`

File: client/components/relatedWidget/ProductListItem.jsx [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+  render () {
+    // console.log('image:', this.props.image);
+    return (<div
+      onClick={()=>{ this.props.getNewProduct(this.props.text.id); }}
+      className={`menu-item ${this.props.selected ? 'active' : ''}`}
+      >
+        <h4>{this.props.text.category}</h4>
+        <h4>{this.props.text.name}</h4>
+        <img src={this.props.image} width='200' height='250'/>
```
Comment (cchauche) [inline client/components/relatedWidget/ProductListItem.jsx:17]: Don't forget to add an 'alt' attribute for accessibility, could probably just be the product name.

File: client/components/relatedWidget/RelatedCarousel.jsx [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+          .then(results=>{
+            return results.map((res)=>{
+              return res.data.results[0].photos[0].thumbnail_url;
+            });
+          })
+          .then(results=>{ this.setState({relatedImages: results}); })
+          .catch(err=>{ console.log(err); });
+    })
+    .catch(err=>{ console.log(err); });
+  }
```
Comment (cchauche) [inline client/components/relatedWidget/RelatedCarousel.jsx:76]: I'm finding this function difficult to follow, I'm also curious why you are making a new getRelatedProduct function here when there is already one at the App level?

File: client/components/relatedWidget/RelatedCarousel.jsx [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+class RelatedCarousel extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      selected: selected,
+      relatedIds: [],
+      relatedProducts: [],
+      relatedImages: []
+    };
+    this.productItems = ProductList(
```
Comment (cchauche) [inline client/components/relatedWidget/RelatedCarousel.jsx:26]: Interesting, are you basically adding an array of React component as a property on this class constructor?
So this.productItems would look something like: `[ <ProuctListItem />, <ProuctListItem />, <ProuctListItem /> ]`

File: client/components/relatedWidget/RelatedCarousel.jsx [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+      this.state.selected,
+      this.props.getNewProduct);
+      // console.log('inside buildCarousel');
+  }
+
+  componentDidMount() {
+    // console.log('componentDidMount');
+    // console.log(this.props.product)
+    this.buildCarousel();
+  }
```
Comment (cchauche) [inline client/components/relatedWidget/RelatedCarousel.jsx]: It looks like buildCarousel does the same thing as the initial state of `this.productItems`
If that is true then you immediately replace `this.productItems` when `componentDidMount()` runs.  Any reason you can't just set the initial value of `this.productItems` to `[]` and that way remove the duplicate logic and the extra calculations?

File: public/styles.css [hunk 8]
```diff
@@ -4,3 +4,22 @@ body h1 {
...
+}
+
+.scroll-menu-arrow {
+  padding: 20px;
+  cursor: pointer;
+}
+
+.scroll-menu-arrow--disabled {
+  visibility: hidden;
+}
```
Comment (cchauche) [inline public/styles.css:25]: Should this file be added to .gitignore since its created by webpack?

**Review by cchauche — APPROVED** [review]

**Comment by cchauche** [comment]
Sorry I didn't leave you a comment about the module in general which is that I'm super stoked that the carousel basically works.  It was super exciting to click on an image and watch everything change so good work!  I left you a bunch of comments but I can see the general thing works so I've approved it

**Review by rhunce — APPROVED** [review]

**Pull request merged**